sdk/tests/types/address/mod.rs

```diff
@@ -59,3 +68,152 @@ fn is_valid_bech32() {
         "rms1qpllaj0pyveqfkwxmnngz2c488hfdtmfrj3wfkgxtk4gtyrax0jaxzt70zY"
     ));
 }
+
+#[test]
+fn capabilities() {
+    let address = RestrictedAddress::new(rand_ed25519_address())
+        .unwrap()
+        .with_allowed_capabilities([0])
+        .unwrap();
+    let mut capabilities = address.allowed_capabilities()[0];
+
+    assert!(!capabilities.has_capabilities(CapabilityFlag::NATIVE_TOKENS));
+    capabilities.add_capabilities(CapabilityFlag::NATIVE_TOKENS);
+    assert!(capabilities.has_capabilities(CapabilityFlag::NATIVE_TOKENS));
+
+    assert!(!capabilities.has_capabilities(CapabilityFlag::MANA));
+    capabilities.set_capabilities(CapabilityFlag::MANA | CapabilityFlag::DELEGATION_OUTPUTS);
+    assert!(capabilities.has_capabilities(CapabilityFlag::MANA));
+    assert!(capabilities.has_capabilities(CapabilityFlag::DELEGATION_OUTPUTS));
+    assert!(!capabilities.has_capabilities(CapabilityFlag::NATIVE_TOKENS));
+}
+
+#[test]
+fn ed25519_bech32() {
+    // Test from https://github.com/iotaledger/tips-draft/blob/tip50/tips/TIP-0050/tip-0050.md#bech32-strings
+    let address = Ed25519Address::new(
```

Comment:
`Ed25519Address::from_public_key_bytes` sounds like an interesting method to have

Reply:
added